homeassistant/components/pi_hole/config_flow.py

```diff
@@ -136,22 +97,38 @@ async def async_step_init(
             errors=errors,
         )
 
-    async def async_step_api_key(
-        self, user_input: dict[str, Any] | None = None
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        self._config = self.init_data
```

Comment:
Where is `init_data` defined?

Reply:
it comes from here

https://github.com/home-assistant/core/blob/77dd0b2fd3da885571f4d78f4ab23b6531d2b3c3/homeassistant/data_entry_flow.py#L441

TBH i'm always struggling were or how to access the configuration entry is question in case an reauth flow is started 🤔

Reply:
Just use the `entry_data` parameter.

`init_data` is not normally used in the flows themselves, but only in flow helpers.

Reply:
i'm not sure, why ´entry_data` did not work for me in my tests before, but now it does 🤷‍♂️  -> [`05607b8`](https://github.com/home-assistant/core/pull/85264/commits/05607b8d68ef5ccfccc19cb201e66fe1aefef3b7)